File: app/src/main/java/app/vodio/com/vodio/beans/Vod.java
package app.vodio.com.vodio.beans;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

public class Vod {
    float timeInSecond;
    String authorLogin;
    String title;
    List<String> tags;
    String audioFilePath;

    public Vod(JSONObject obj){
        try {
            timeInSecond = Float.parseFloat(obj.get("timeInSecond").toString());
            authorLogin = obj.get("authorLogin").toString();
            title = obj.get("title").toString();
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public float getTimeInSecond() {
        return timeInSecond;
    }

    public void setTimeInSecond(float timeInSecond) {
        this.timeInSecond = timeInSecond;
    }

    public String getAuthorLogin() {
        return authorLogin;
    }

    public void setAuthorLogin(String authorLogin) {
        this.authorLogin = authorLogin;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<String> getTags() {
        return tags;
    }

    public void setTags(List<String> tags) {
        this.tags = tags;
    }

    public String getAudioFilePath() {
        return audioFilePath;
    }

    public void setAudioFilePath(String audioFilePath) {
        this.audioFilePath = audioFilePath;
    }
}

File: app/src/main/java/app/vodio/com/vodio/beans/Vod.kt
package app.vodio.com.vodio.beans

import android.media.MediaMetadataRetriever
import java.io.File
import java.util.*
import java.util.concurrent.Future

data class Vod(var idVod : Int = -1, var timeInSecond: Int? = 0, var authorLogin: String = "",
                    var title: String = "", var tags: List<String> = ArrayList<String>(), var audioFilePath: String? = "",
                        var futureFile : Future<File>? = null, var timestamp: Long = Calendar.getInstance().timeInMillis
                            , var likedByFollower : Int = 0)
    {
        private var file : File? = null
        private var isLocal = false
        fun isLikedByFollower() : Boolean{
            return likedByFollower == 1
        }

        fun setFile(f : File){
            file = f
            if(timeInSecond == 0) {
                val mmr = MediaMetadataRetriever()
                mmr.setDataSource(file!!.path)
                val durationStr = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)
                timeInSecond = Integer.parseInt(durationStr) / 1000
            }
        }

        fun getFile() : File? {
            if(futureFile != null && futureFile!!.isDone){
                file = futureFile!!.get()
                futureFile = null
            }
            return file
        }

        fun initTimestamp(){
            timestamp = Calendar.getInstance().timeInMillis
        }

        fun setIsLocal(b : Boolean){
            isLocal = b
        }


        fun isLocal() : Boolean{
            return isLocal
        }

}
